__tests__/app.test.js

```diff
@@ -0,0 +1,72 @@
+const app = require("../app.js");
+const request = require("supertest");
+const db = require("../db/connection.js");
+const seed = require("../db/seeds/seed.js");
+const data = require("../db/data/test-data/index");
+const devData = require("../db/data/development-data/index.js");
+
+const { toLocaleString } = require("../db/data/development-data/articles.js");
+beforeAll(() => seed(data));
+afterAll(() => db.end());
+const availableRoutes = require("../app.js");
+
+describe("GET /api/topics", () => {
+  test("returns 200 status code", () => {
+    return request(app).get("/api/topics").expect(200);
+  });
+
+  test("returns an array of topic objects", () => {
+    return request(app)
+      .get("/api/topics")
+      .then(({ body, status }) => {
+        if (status === 404) {
+          console.log("Error", status);
+        }
+        if (status === 500) {
+          console.log("Internal server error", status);
+        }
+        body.topics.forEach((topic) => {
+          expect(topic).toHaveProperty("description");
+          expect(topic).toHaveProperty("slug");
+          expect(typeof topic).toBe("object");
+          expect(typeof topic.description).toBe("string");
+          expect(typeof topic.slug).toBe("string");
```

Comment:
Whenever you request multiple resources from your API you need to make sure that the right **number** of resources come back. This means you need to do a length check as well as the forEach on body.topics.

Lines 29-30 are also redundant, this is because on line 32-33 you HAVE to have the description and slug properties on the topic for those lines to work. You can get rid of the toHaveProperty checks in this test as a result.

Reply:
thanks, i have added the length test, is it ok to keep lines 29 and 30? i know the other tests also prove they exist but its useful for me to remember we can test properties like that, if not i will remove 

expect(body.topics.length).toBe(3);
        body.topics.forEach((topic) => {
          expect(topic).toHaveProperty("description");
          expect(topic).toHaveProperty("slug");
          expect(typeof topic).toBe("object");
          expect(typeof topic.description).toBe("string");
          expect(typeof topic.slug).toBe("string");
        });